plotly/utils.py

```diff
@@ -399,3 +404,46 @@ def get_by_path(obj, path):
     for key in path:
         obj = obj[key]
     return obj
+
+
+### validation
+def validate_world_readable_and_sharing_settings(option_set):
+    if ('world_readable' in option_set and
+        option_set['world_readable'] is not None and
+        option_set['world_readable'] is True and
+        'sharing' in option_set and
+        option_set['sharing'] is not None and
+            option_set['sharing'] != 'public'):
+        raise PlotlyError(
+            "Looks like you are setting your plot privacy to both "
+            "public and private.\n If you set world_readable as True, "
+            "sharing can only be set to 'public'")
+    elif ('world_readable' in option_set and
+          option_set['world_readable'] is False and
+          'sharing' in option_set and
+          option_set['sharing'] == 'public'):
+        raise PlotlyError(
+            "Looks like you are setting your plot privacy to both "
+            "public and private.\n If you set world_readable as "
+            "False, sharing can only be set to 'private' or 'secret'")
+    elif ('sharing' in option_set and
+          option_set['sharing'] not in ['public', 'private', 'secret', None]):
+        raise PlotlyError(
+            "The 'sharing' argument only accepts one of the following "
+            "strings:\n'public' -- for public plots\n"
+            "'private' -- for private plots\n"
+            "'secret' -- for private plots that can be shared with a "
+            "secret url"
+        )
```

Comment:
:cow2: This logic is hard to read though. I didn't fill out the exact messages, but would something like this work?

```
world_readable = option_set.get('world_readable')
sharing = option_set.get('sharing')

if 'world_readable' in option_set and world_readable not in [True, False]:
    raise PlotlyError('The world_readable option must be True or False.')

if 'sharing' in option_set and sharing not in ['public', 'private', 'secret']:
    raise PlotlyError("The sharing option must be 'public', 'private', or 'secret'.")

if world_readable is True and sharing != 'public':
    raise PlotlyError("If world_readable is True, sharing must be 'public'.")

if world_readable is False and sharing == 'public':
    raise PlotlyError("If world_readable is False, sharing cannot be 'public'.")
```

Just a cow tip, take it er leave it :) I didn't actually test this.